bin/compile

```diff
@@ -7,10 +7,21 @@ CACHE_DIR=$2
 ENV_DIR=$3
 
 LIBMAXMINDDB_VERSION=1.2.1
-GEOLITE2_CITY_FILENAME="GeoLite2-City.mmdb"
-GEOLITE2_CITY_TARBALL_FILENAME="GeoLite2-City.tar.gz"
-GEOLITE2_COUNTRY_FILENAME="GeoLite2-Country.mmdb"
-GEOLITE2_COUNTRY_TARBALL_FILENAME="GeoLite2-Country.tar.gz"
+
+# Use ENV var and fall back to license file
+# Expects ENV var to be `MAXMIND_LICENSE_KEY`
+MAXMIND_LICENSE_FROM_FILE=`cat $ENV_DIR/MAXMIND_LICENSE_KEY`
+MAXMIND_LICENSE=${MAXMIND_LICENSE_KEY:=$MAXMIND_LICENSE_FROM_FILE}
```

Comment:
@albertxie and I initially thought that configuration variables were inherently available to the build phase for buildpacks, but after reading the [API documentation](https://devcenter.heroku.com/articles/buildpack-api), we realize this is not the case.

Instead, as you can see in [this section](https://devcenter.heroku.com/articles/buildpack-api#bin-compile-summary), all configuration variables are made accessible to the buildpack through the `$ENV` directory, and are individually injected into separate files. This is where `cat $ENV_DIR/MAXMIND_LICENSE_KEY` comes from.

We initially saw this being used [here](https://github.com/sre-dev/heroku-buildpack-maxmind/blob/master/bin/compile#L15). I'm not sure if this still complies with the 12-factor methodology, but I think since it is injected from a file at build time, it might be.

Reply:
Hm, that looks right. I guess this is why I favour having pull requests and commit history to provide context.

Regardless, this version will work as well, as it falls back to the file based process as well.